src/main/java/com/example/algorithm/QVoterModel.java

```diff
@@ -0,0 +1,148 @@
+package com.example.algorithm;
+
+import com.brunomnsilva.smartgraph.graph.Vertex;
+import com.example.algorithm.operations.ChooseOperation;
+import com.example.algorithm.operations.SendOperation;
+import com.example.algorithm.report.StepReport;
+import com.example.model.MyGraph;
+import com.example.model.MyVertex;
+import com.example.settings.AlgorithmSettings;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+
+import java.util.*;
+
+public class QVoterModel implements Algorithm{
+    private MyGraph<Integer, Integer> graph;
+    private BooleanProperty isFinished = new SimpleBooleanProperty(false);
+    private MyVertex<Integer> selectedAgent;
+    private List<Boolean> opinionsReceived;
+    private QVoterModel.AlgorithmPhase algorithmPhase = QVoterModel.AlgorithmPhase.SEND;
+    private ProbabilityType probabilityType = ProbabilityType.LINEAR;
+    private int q;
+    private int maxTime;
+    private int time = 0;
+
+    @Override
+    public AlgorithmType getType() {
+        return AlgorithmType.QVOTER;
+    }
+
+    @Override
+    public void loadEnvironment(MyGraph<Integer, Integer> graph, AlgorithmSettings settings) {
+        this.graph = graph;
+        maxTime = (int)settings.getSettings().get("time").getValue();
+        q = (int)settings.getSettings().get("q").getValue();
+    }
+
+    @Override
+    public StepReport step() {
+        switch (algorithmPhase){
+            case SEND -> {
+                time ++;
+                algorithmPhase = QVoterModel.AlgorithmPhase.CHOOSE;
+                return sendOpinions();
+            }
+            case CHOOSE -> {
+                algorithmPhase = QVoterModel.AlgorithmPhase.SEND;
+                if(time == maxTime){
+                    isFinished.setValue(true);
+                }
+                return makeDecision();
+            }
+        }
+        return null;
+    }
+
+    private StepReport sendOpinions() {
+        QVoterStepRecord report = new QVoterStepRecord();
+
+        int agentIndex = new Random().nextInt(graph.numVertices());
+        selectedAgent = (MyVertex<Integer>) graph.vertices().stream().toList().get(agentIndex);
+        List<Vertex<Integer>> agentNeighbours = getNeighbours(selectedAgent);
+        report.fillRoles(selectedAgent, agentNeighbours);
+
+        opinionsReceived = new ArrayList<>();
+        for(Vertex<Integer> neighbour : agentNeighbours){
+            BooleanProperty opinion = ((MyVertex<Integer>) neighbour).getNextOpinion(selectedAgent);
+            opinionsReceived.add(opinion.getValue());
+            report.getOperations().add(new SendOperation(neighbour, selectedAgent, opinion));
+        }
+
+        return report;
+    }
+
+    private StepReport makeDecision() {
+        QVoterStepRecord report = new QVoterStepRecord();
+
+        report.fillRoles(selectedAgent, null);
+
+        if(opinionsReceived.stream().distinct().count() <= 1 ||  checkProbability()){
```

Comment:
For readability I would extract this condition to method called e.g. shouldAcceptNeighboursOppinion().